corehq/apps/users/forms.py

```diff
@@ -1234,17 +1234,25 @@ def save(self, user):
                 updated_program_id = program_id
             domain_membership.program_id = program_id
 
-        location_updates = self._update_location_data(user)
+        new_location_id = self.cleaned_data['primary_location']
+        new_location_ids = self.cleaned_data['assigned_locations']
+        error_response = self._validate_location_data(new_location_id, new_location_ids)
+        if error_response:
+            return error_response
```

Comment:
Can this go in `clean`?  That seems more idiomatic, and will allow the view to see that the form was invalid.  Does it presently anticipate a response from `form.save()`?